docs/proposals/002-api-proposal/proposal.md

```diff
@@ -0,0 +1,254 @@
+ - Manage Pod lifecycle of pods within the pool 
+
+Additionally, any Pod that seeks to join a BackendPool would need to support a protocol, defined by LLM Instance Gateway, to ensure the Pool has adequate information to intelligently route requests.
+
+### LLMUseCase
+
+An LLMUseCase allows the UseCaseOwner to define:
+- Which LoRA adapter(s) to consume 
+  - LLMUseCase allows for traffic splitting between adapters _in the same pool_ to allow for new LoRA adapter versions to be easily rolled out 
+- SLO objectives for the UseCAse
+- The Pools this UseCase is relevant to 
+
+### Spec
+
+**BackendPool**
+```golang
+// A grouping of model serving instances that are expected to serve the same model(s) and speak the same API protocol
+// The LLMBackendPool is also the scope for enforcing priority and fairness across different use cases.
+// When used with the Gateway API, the LLMBackendPool can serve as the BackendRefs of an HTTPRoute.
+// Generally the LLMBackendPool is owned by the "Inference Platform Admin" persona.
+type LLMBackendPool struct {
+        metav1.ObjectMeta
+        metav1.TypeMeta
+        Spec LLMBackendSpec
+}
+
+type LLMBackendPoolSpec struct {
+        // Pod selector, similar to k8s Service.
+        Selector map[string]string `json:"selector,omitempty"`
+        // Allows names within the `model` param that don't have an explicitly defined UseCase to pass through. Defaults to `false`.
+        AllowUndefinedModels bool
+        // Admin-defined minimum objective value that can be requested, will cause an error in LLMUseCase if limit is broken.
+        MinTPOT float32
+}
+```
+
+
+**LLMUseCase**
+```golang
+// LLMUseCase represents a use case of an LLM, which is multiplexed onto one or more LLMBackendPools.
+// It allows mapping a use case to backend pools, and models or adapters that backend model servers understand.
+// The LLMUseCase defines request routing behavior within an LLMBackendPool.
+// Generally the LLMUseCase is owned by the "LLM Use Case Owner" persona, which can be teams in an organization.
+type LLMUseCase struct {
+        metav1.ObjectMeta
+        metav1.TypeMeta
+
+        Spec LLMUseCaseSpec
+}
+
+type LLMUseCaseSpec struct {
+        // Map use case to one or more backend pools.
+        // In the most common case, this should be a single backend pool.
+        // Multiple backend pools can be used for traffic splitting while migrating to a new backend pool.
+        Rules   []LLMUseCaseRule
+}
+
+// LLMUseCaseRule represents a mapping from a LLMUseCase to a backend pool and adapters/models in that pool.
+type LLMUseCaseRule struct {
+        // The name used in the `model` param of incoming requests
+        ModelName string
+        // Optional
+        Objective *Objective
+        // Required.
+        // Reference to an LLMBackendPool. This allows registering a use case
+        // as valid on a pool. 
+        // NOTE: Allowing multiple pools is a configuration convenience.
```

Comment:
What's the expected behavior when multiple pools are specified?

Reply:
Yeah, so absolutely open to feedback here. The multiple pools is not a behavior design choice, but one of convenience. 

For instance if a user wants to specify the exact same usecase across multiple pools, they just add to this list, rather than having a set of different objects.

Reiterating that feedback here is appreciated, I do think it's potentially confusing, but valuable for power-usage